src/shared/components/blast-sequence-button/BlastSequenceButton.tsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { useNavigate } from 'react-router';
+import classNames from 'classnames';
+
+import * as urlFor from 'src/shared/helpers/urlHelper';
+
+import useBlastForm from 'src/content/app/tools/blast/hooks/useBlastForm';
+
+import { useAppDispatch } from 'src/store';
+
+import {
+  addSelectedSpecies,
+  clearBlastForm
+} from 'src/content/app/tools/blast/state/blast-form/blastFormSlice';
+
+import { ImageButton } from 'src/shared/components/image-button/ImageButton';
+import { BlastIcon } from 'src/shared/components/app-icon';
+
+import type { Species } from 'src/content/app/tools/blast/state/blast-form/blastFormSlice';
+import type { SequenceType } from 'src/content/app/tools/blast/types/blastSettings';
+
+import styles from './BlastSequenceButton.scss';
+
+// TODO: think about the possible header for the sequence
+type Props = {
+  label?: string;
+  species: Species;
+  sequence?: string; // if the sequence isn't provided, the button will be disabled
+  sequenceType: SequenceType;
```

Comment:
should this not be optional as well if `sequence` is optional?

Reply:
I think the parent that is using this component will always know the sequence type, even if it may be fetching the sequence.

If this changes, we may change this prop later.

Reply:
I am confused... so even if `props.sequence` isn't provided as mentioned in the comment above, we still want to know about the sequenceType?

Reply:
So far, I've only seen that the parent component may not know about the sequence (while it is still fetching it), but has a pretty clear idea what it is that it is fetching.

I do not know which other data can be legitimately unavailable to the parent that wants to render the blast button.